code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -1336,3 +1336,837 @@
+	var/obj/item/organ/internal/vaurca/neuralsocket/host = src.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(src.stat != CONSCIOUS)
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
+		return
+	if(stat!=CONSCIOUS)
+		remoteview_target = null
+		reset_view(0)
+		return
+	if(client.eye != client.mob)
+		remoteview_target = null
+		reset_view(0)
+		return
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before attempting another Hivenet action!"))
+	for(var/mob/living/carbon/human/player in living_mob_list)
+		if(isvaurca(player) && player.internal_organs_by_name[BP_NEURAL_SOCKET])
+			if(player.stat != CONSCIOUS)
+				continue
+			LAZYADD(available_vaurca, player)
+	LAZYREMOVE(available_vaurca, src) //you already have the supernatural ability to see through your own eyes
+	var/mob/living/carbon/human/target = input(src, "Select a Vaurca to observe.", "Hivenet Remote Observation") as null|anything in available_vaurca
+	if(!target || !isvaurca(target))
+		remoteview_target = null
+		reset_view(0)
+		return
+	var/obj/item/organ/internal/vaurca/neuralsocket/S = target.internal_organs_by_name[BP_NEURAL_SOCKET]
+	var/list/target_fullname = splittext(target.name, " ")
+	var/target_surname = target_fullname[2]
+	if(target_surname == surname && !S.shielded)
+		to_chat(src, SPAN_NOTICE("You extend your mind into the Hivenet, accessing [target]'s senses. Use this verb again to cancel."))
+		remoteview_target = target
+		reset_view(target)
+		return
+	else
+		var/choice = alert(target, "[src] is attempting to access your senses. Do you wish to allow this?", "Hivenet Remote Observation", "Deny", "Allow")
+		if(choice == "Allow")
+			to_chat(src, SPAN_NOTICE("You extend your mind into the Hivenet, accessing [target]'s senses. Use this verb again to cancel."))
+			remoteview_target = target
+			reset_view(target)
+			return
+		if(choice == "Deny")
+			if(istype(src.origin, /singleton/origin_item/origin/mouv_b))
+				var/hijack = alert(src, "[target] has denied your access request. Attempt a hijack?", "Hijack Hivenet Senses", "No", "Yes")
+				if(hijack == "Yes")
+					if(prob(40) && !S.shielded)
+						to_chat(src, SPAN_NOTICE("You extend your mind into the Hivenet, accessing [target]'s senses. Use this verb again to cancel."))
+						remoteview_target = target
+						reset_view(target)
+						return
+					else
+						to_chat(src, SPAN_WARNING("You attempt a hijack, and a stab of pain comes to your mind as your attempt is rejected!"))
+						src.adjustHalLoss(15)
+						src.flash_pain(15)
+						to_chat(target, SPAN_WARNING("[src] attempted to hijack your senses, but your countermeasures repelled them!"))
+						remoteview_target = null
+						reset_view(0)
+						host.last_action = world.time + 5 MINUTES
```

Comment:
The cooldown needs to be set before you get to this big if, otherwise you can kind of spam someone with popups